user/enterprise/custom-queues.md

```diff
@@ -0,0 +1,89 @@
+
+<div id="toc"></div>
+
+## Enable Queues on the Platform
+
+To allow your Travis CI Enterprise platform instance to route jobs to customized queues, the `template_selection` and `multi_os` feature flags must be set. To do this, ssh into your platform server, then run `travis console`. Run the following command to enable the required feature flags: 
+```
+Travis::Features.enable_for_all(:template_selection); Travis::Features.enable_for_all(:multi_os); exit
+```
+
+The new settings will take effect immediately, but job routing will remain the same until new queues are defined. 
+
+## Define Custom Queues in the Management Console  
+
+Once the custom queue feature flags are enabled, the routing must configured in the management console. This is defined in yaml, in the **Advanced Configuration YAML** section at the bottom of the management console **Settings** page, e.g. `https://<your-domain>:8800/settings`.  
+
+There are a number of options/selectors used to define routing to a custom queue. Some must be specified in the `.travis.yml`, others (such as those that indicate repo ownership) should not be specified in the `.travis.yml`. Repos that match _all_ of the selectors for a custom queue will be built on that custom queue. We recommend using the following selectors:
+
+  - `language` - defines the build environment based on the chosen language of the project†
+  - `group` - mostly semantic, for defining 'groups' of environments†
+  - `owner` - either be an org or a user
+  - `slug` - a repository, in the form: `org/repo` or `user/repo`
+
+
+† `language` and `group` selectors must be specified in the `.travis.yml`. See [the example](#Advanced-Configuration-YAML-Example) for more details. 
+
+Note: We do not recommend using `dist` and `os` for these selectors. These two have some of their own routing processes built-in and may not entirely behave as intended. 
+
+All selectors should be given as a list in the "Advanced Configuration YAML" in the format:
+```yaml
+production:
+  queues:
+  - queue: name
+    selector: value
+  - queue: another.queue
+    selector: different_value
+    selector: something_else
+```
+see the [example](#Advanced-Configuration-YAML-Example) for details on syntax. Click "Save" on the Management Console Settings when you are ready. Travis CI Enterprise will restart, with your new queue settings.
+
+### Advanced Configuration YAML Example 
+
+The syntax for the **Advanced Configuration YAML** field is very important. Improper syntax will result in builds being routed to defaults, usually a `builds.linux` queue (if this has not been modified on your installation.) Syntax for a custom queue looks as follows: 
```

Comment:
lol at Improper. Maybe incorrect? 

looks as follows reads a little odd to me, how about "here is an example custom queue definition" or something?

Reply:
but... but... it's improper. It's not what Proper Yaml Writers use.

Agreed - "incorrect" is good 👍  Also, I like that wording on "example"